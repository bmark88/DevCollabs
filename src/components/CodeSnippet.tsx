import React, { ReactNode, useEffect } from "react"
import AceEditor from "react-ace";

import "ace-builds/src-noconflict/mode-javascript";
import "ace-builds/src-noconflict/theme-twilight";

interface Props {
  children: ReactNode
}

const CodeSnippet = ({ children } :Props) => {
  function onChange(newValue) {
    console.log("change", newValue);
  }

  return (
  <AceEditor
    mode="javascript"
    theme="twilight"
    onChange={onChange}
    name="UNIQUE_ID_OF_DIV"
    editorProps={{ $blockScrolling: true }}
  />
    )
};

export default CodeSnippet;