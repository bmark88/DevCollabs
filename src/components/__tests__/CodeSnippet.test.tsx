import React from "react"
import renderer from "react-test-renderer"
// import { render } from "@types/testing-library__react"

import {
    render,
    cleanup,
    waitForElement,
    fireEvent,
    getByText,
    getAllByTestId,
    getByAltText,
    getByPlaceholderText,
    queryByText,
    queryByAltText,
  } from "@testing-library/react"


import CodeSnippet from "../CodeSnippet"

afterEach(cleanup)

describe("CodeSnippet", () => {
  xit("renders correctly", async () => {
    
    const tree = renderer
      .create(<CodeSnippet />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })


xit("Username displays when someone has entered ", async () => {

})


xit("the value changes in the IDE", async () => {

})


xit("The CAD changes according ", async () => {

})


})