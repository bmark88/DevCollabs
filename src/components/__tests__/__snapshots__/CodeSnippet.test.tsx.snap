// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeSnippet renders correctly 1`] = `
Array [
  <div
    id="UNIQUE_ID_OF_DIV"
    style={
      Object {
        "height": "84.7vh",
        "margin": "1em",
        "minWidth": "500px",
        "width": "520px",
      }
    }
  />,
  <div
    className="sc-AxjAm idqmap"
  >
    Copy to Clipboard:
    <button
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4l6 6v10c0 1.1-.9 2-2 2H7.99C6.89 23 6 22.1 6 21l.01-14c0-1.1.89-2 1.99-2h7zm-1 7h5.5L14 6.5V12z"
        />
      </svg>
    </button>
  </div>,
]
`;
